plugins/lsp/language-servers/pylsp.nix

```diff
@@ -536,14 +536,25 @@ in
           mapAttrsToList
             (
               pluginName: nixPackage:
+              let
+                # Get rid of the python-lsp-server dependency
+                filterDependencies = dependencies: filter (dep: dep.pname != "python-lsp-server") dependencies;
+              in
               (optional (isEnabled plugins.${pluginName}) (
-                nixPackage.overridePythonAttrs (old: {
-                  # Get rid of the python-lsp-server dependency
-                  propagatedBuildInputs = filter (dep: dep.pname != "python-lsp-server") old.propagatedBuildInputs;
-
-                  # Skip testing because those naked dependencies will complain about missing pylsp
-                  doCheck = false;
-                })
+                nixPackage.overridePythonAttrs (
+                  old:
+                  mkMerge [
+                    (mkIf (old ? propagatedBuildInputs) {
+                      propagatedBuildInputs = filterDependencies old.propagatedBuildInputs;
+                    })
+                    (mkIf (old ? dependencies) { dependencies = filterDependencies old.dependencies; })
```

Comment:
This won't work, because `mkIf` is "smart" and evaluates even if the condition is false.

Since (unlike most conditional config) we _don't_ want to evaluate in this case, we should use an `if then else` wrapper such as `lib.optionalAttrs` instead.

Also, I don't think `mkMerge` and `mkIf` can be used in an override function, since they are only intended for use with module config definitions.

Something like this should work:

```nix
(
  old:
  # Note: Depending on the exact nixpkgs rev, we either need to filter `dependencies` or `propagatedBuildInputs`
  # TODO: Only filter `dependencies`
  lib.optionalAttrs (old ? propagatedBuildInputs) (filterDependencies old.propagatedBuildInputs)
  // lib.optionalAttrs (old ? dependencies) (filterDependencies old.dependencies)
)
```

Reply:
Or using `genAttrs`:

```nix
(
  old:
    lib.genAttrs
    (lib.filter (n: old ? ${n}) [
      "propagatedBuildInputs"
      "dependencies"
    ])
    (n: filterDependencies old.${n})
)
```

Reply:
this looks so much better!